presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java

```diff
@@ -212,6 +214,9 @@ public boolean advanceNextPosition()
         }
         catch (IOException | SerDeException | RuntimeException e) {
             closeWithSuppression(this, e);
+            if (e instanceof TextLineLengthLimitExceededException) {
+                throw new PrestoException(HIVE_BAD_DATA, e);
```

Comment:
Let's add a more user friendly error message:
```java
throw new PrestoException(HIVE_BAD_DATA, "Line too long in text file: " + path, e);
```
You'll need to pass the path into the constructor.